HW2/P2/P2.py

```diff
@@ -40,12 +42,14 @@
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
-    N = 10
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained uncorrelated: {} seconds".format(t.interval))
+    T1 = []
+    for N in Nrange:
+        counts[:] = orig_counts
+        with Timer() as t:
+            move_data_medium_grained(counts, src, dest, 100, N)
+        assert counts.sum() == total, "Wrong total after move_data_medium_grained"
+        T1.append(t.interval)
+        print("Medium grained uncorrelated N={}: {} seconds".format(N, t.interval))
```

Comment:
Nice, pretty similar to how I did it for this part -- one difference was that I did a slightly more fine-grained search over a smaller space (i.e. I went over N = 2 to 30 by 2s) rather than the array you chose but I think either way is fine!